src/ConnectionAutomaton.cpp

```diff
@@ -182,6 +183,16 @@ void ConnectionAutomaton::onConnectionFrame(Payload payload) {
       // TODO handle lease logic
       LOG(INFO) << "ignoring setup frame";
       return;
+    case FrameType::METADATA_PUSH: {
+      Frame_METADATA_PUSH frame;
+      if (frame.deserializeFrom(std::move(payload))) {
+        assert(frame.header_.flags_ & FrameFlags_METADATA);
```

Comment:
This logic with flipping the flag and responding is particular to keepalive

> Reception of a KEEPALIVE frame with the (R)espond flag set MUST cause a client or server to send back a KEEPALIVE with the (R)espond flag NOT set. The data in the received KEEPALIVE MUST be echoed back in the generated KEEPALIVE.

For metadata push, we need to allow someone implementing RequestHandler to receive the metadata payload.  e.g. compare 

https://github.com/ReactiveSocket/reactivesocket-java/blob/8408978e638c7e42948017d0e74a37ce115c2aec/reactivesocket-core/src/main/java/io/reactivesocket/RequestHandler.java#L54

and

https://github.com/ReactiveSocket/reactivesocket-cpp/blob/2be2f4f5e74c91506f6be680e57290bfbb214dd4/src/RequestHandler.h